ceph-releases/luminous/ubuntu/16.04/daemon/zap_device.sh

```diff
@@ -1,8 +1,102 @@
 #!/bin/bash
 set -e
 
+# Get a list of child devices from a root device
+function get_child_partitions {
+# $1: raw parent device
+  if [ ! -e  "${1}" ] && [ ! -b "${1}" ]; then
+    log "Error, ${1} doesn't seem to be a valid device!"
+    exit 1
+  fi
+  local parts
+  parts=$(lsblk -no KNAME "${1}")
+  for p in $parts; do
+    kname=$(lsblk --nodeps -no KNAME /dev/"${p}")
+    pkname=$(lsblk --nodeps -no PKNAME /dev/"${p}")
+    if [ "${kname}" != "${pkname}" ] && [ -n "${pkname}" ]; then
+      echo /dev/"${p}"
+    fi
+  done
+}
+
+function get_dmcrypt_uuid_part {
+  # look for Ceph encrypted partitions
+  # Get all dmcrypt for ${device}
+  blkid -t TYPE="crypto_LUKS" "${1}"* -o value -s PARTUUID
```

Comment:
still getting confused by this one.

Reply:
ok, got it :)
ideally I guess we could enhance `get_part_uuid` but I guess it's just too complex.